code/modules/modular_computers/hardware/network_card.dm

```diff
@@ -28,92 +27,84 @@ var/global/ntnet_card_uid = 1
 		. += "OpenEth (Physical Connection) - Physical network connection port"
 
 /obj/item/stock_parts/computer/network_card/Initialize()
-	. = ..()
-	identification_id = ntnet_card_uid
-	ntnet_card_uid++
+	if(!identification_id)
+		identification_id = new_guid()
+	set_extension(src, /datum/extension/exonet_device, ennid, get_netspeed())
+	return INITIALIZE_HINT_LATELOAD
+
+/obj/item/stock_parts/computer/network_card/LateInitialize()
+	if(ennid)
+		set_ennid(ennid, keydata)
 
 /obj/item/stock_parts/computer/network_card/advanced
-	name = "advanced NTNet network card"
-	desc = "An advanced network card for usage with standard NTNet frequencies. It's transmitter is strong enough to connect even when far away."
+	name = "advanced EXONET network card"
+	desc = "An advanced network card for usage with standard EXONET frequencies. It's transmitter is strong enough to connect even when far away."
 	long_range = 1
 	origin_tech = "{'" + TECH_DATA + "':4,'" + TECH_ENGINEERING + "':2}"
 	power_usage = 100 // Better range but higher power usage.
 	icon_state = "netcard_advanced"
 	hardware_size = 1
-	matter = list(MAT_STEEL = 500, MAT_GLASS = 200)
 
 /obj/item/stock_parts/computer/network_card/wired
-	name = "wired NTNet network card"
-	desc = "An advanced network card for usage with standard NTNet frequencies. This one also supports wired connection."
+	name = "wired EXONET network card"
+	desc = "An advanced network card for usage with standard EXONET frequencies. This one also supports wired connection."
 	ethernet = 1
 	origin_tech = "{'" + TECH_DATA + "':5,'" + TECH_ENGINEERING + "':3}"
 	power_usage = 100 // Better range but higher power usage.
 	icon_state = "netcard_ethernet"
 	hardware_size = 3
-	matter = list(MAT_STEEL = 2500, MAT_GLASS = 400)
```

Comment:
Unintentional removal here and Destroy() below?

Reply:
Destroy removal was appropriate. It is not needed.